src/main/java/mateacademy/onlinebookstore/controller/CategoryController.java

```diff
@@ -0,0 +1,76 @@
+package mateacademy.onlinebookstore.controller;
+
+import io.swagger.v3.oas.annotations.Operation;
+import jakarta.validation.Valid;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import mateacademy.onlinebookstore.dto.book.BookDtoWithoutCategoryIds;
+import mateacademy.onlinebookstore.dto.category.CategoryRequestDto;
+import mateacademy.onlinebookstore.dto.category.CategoryResponseDto;
+import mateacademy.onlinebookstore.service.book.BookService;
+import mateacademy.onlinebookstore.service.category.CategoryService;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("api/categories")
+public class CategoryController {
+    private final CategoryService categoryService;
+    private final BookService bookService;
+
+    @PostMapping
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @Operation(description = "Create a new category")
```

Comment:
Better use ResponseStatus.CREATED  annotation (the same for other endpoints) that create objects